services/movie_session.py

```diff
@@ -0,0 +1,54 @@
+from _datetime import datetime
+from django.db.models import QuerySet
+from typing import Optional
+
+from db.models import MovieSession, CinemaHall, Movie
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    movie_session = MovieSession(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+    movie_session.save()
+
+
+def get_movies_sessions(session_date: Optional[str | None] = None) -> QuerySet:
+    sessions = MovieSession.objects.all()
+
+    if session_date:
+        date_session = datetime.strptime(session_date, "%Y-%m-%d").date()
+        sessions = sessions.filter(show_time__date=date_session)
+
+    return sessions
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime | None] = None,
+        movie_id: Optional[int | None] = None,
+        cinema_hall_id: Optional[int | None] = None
+) -> None:
+    session = MovieSession.objects.get(id=session_id)
+
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie = Movie.objects.get(id=movie_id)
+    if cinema_hall_id:
+        session.cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
```

Comment:
In order to not make 2 additional sql queries for movie and hall, you can set movie_id and cinema_hall_id directly. You can do it despite you haven't these fields in your model because Django is developed in the way that in the code you can use your added fields(movie, cinema_hall) and they contain related objects but under the hood in database ID is saved and column name is movie_id and cinema_hall_id because database doesn't know what is Django, related object but knows what is id